controllers/helmrelease_controller.go

```diff
@@ -65,6 +67,11 @@ import (
 	"github.com/fluxcd/helm-controller/internal/util"
 )
 
+var (
+	targetPathMaxLen = 250
+	targetPathRegex  = regexp.MustCompilePOSIX(`^([a-zA-Z0-9_.\\\/]|\[[0-9]{1,5}\])+$`)
```

Comment:
We could set these in the CRD as kubebuilder annotations, for the Kubernetes API to perform the validation instead of us.

Reply:
Updated the PR accordingly. The only downside of this approach is that existing objects which are made invalid by the new CRD will need to be abide by the new validation before the API Server allows them to be deleted.

That being said, the new rules are largely to formalise the expected format and length, and are not expected to break existing setups.